cfme/tests/services/test_dialog_regex_validation_in_catalog.py

```diff
@@ -66,10 +72,23 @@ def test_dialog_element_regex_validation(appliance, dialog_cat_item):
     catalog_item, ele_name = dialog_cat_item
     service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
     view = navigate_to(service_catalogs, 'Order')
-    view.fields(ele_name).fill("!@#%&")
-    wait_for(lambda: view.submit_button.disabled, timeout=7)
-    view.fields(ele_name).fill("test_123")
-    wait_for(lambda: not view.submit_button.disabled, timeout=7)
+
+    for value in ["test_123", "!@#$%", ""]:
+        view.fields(ele_name).fill(value)
+        assert view.fields(ele_name).read() == value
+
+        if value in ["!@#$%"]:
+            expected_warning = 'Entered text should match the format: ^([a-z0-9]+_*)*[a-z0-9]+$'
+            assert view.fields(ele_name).input.warning == expected_warning
+        try:
+            wait_for(
+                lambda: view.submit_button.disabled,
+                fail_condition=False if value == "!@#$%" else True,
+                timeout=10,
+                delay=2
+            )
+        except TimedOutError:
+            pytest.fail("'{}' fail to match regex validation".format(value))
```

Comment:
I don't think this failure message is accurate - it failed to detect the correct submit_button state.

The failure of matching regex validation happens above in the assert statements against the warning message.